src/System.Net.WebSockets/src/System/Net/WebSockets/WebSocketReceiveResult.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics.Contracts;
+
+namespace System.Net.WebSockets
+{
+    public class WebSocketReceiveResult
+    {
+        public WebSocketReceiveResult(int count, WebSocketMessageType messageType, bool endOfMessage)
+            : this(count, messageType, endOfMessage, null, null)
+        {
+        }
+
+        public WebSocketReceiveResult(int count,
+            WebSocketMessageType messageType,
+            bool endOfMessage,
+            WebSocketCloseStatus? closeStatus,
+            string closeStatusDescription)
+        {
+            if (count < 0)
+            {
+                throw new ArgumentOutOfRangeException("count");
+            }
+
+            Count = count;
+            EndOfMessage = endOfMessage;
+            MessageType = messageType;
+            CloseStatus = closeStatus;
+            CloseStatusDescription = closeStatusDescription;
+        }
+
+        public int Count { get; private set; }
+        public bool EndOfMessage { get; private set; }
+        public WebSocketMessageType MessageType { get; private set; }
+        public WebSocketCloseStatus? CloseStatus { get; private set; }
+        public string CloseStatusDescription { get; private set; }
+
+        internal WebSocketReceiveResult Copy(int count)
+        {
+            Contract.Assert(count >= 0, "'count' MUST NOT be negative.");
+            Contract.Assert(count <= Count, "'count' MUST NOT be bigger than 'this.Count'.");
```

Comment:
For asserts, across corefx we've been trying to standardize on Debug.Assert rather than Contract.Assert.